packages/javascript/bh-shared-ui/src/views/UserProfile/UserProfile.tsx

```diff
@@ -46,8 +46,19 @@ const UserProfile = () => {
     );
 
     const updateUserPasswordMutation = useMutation(
-        ({ userId, secret, needsPasswordReset }: { userId: string; secret: string; needsPasswordReset: boolean }) =>
+        ({
+            userId,
+            secret,
+            needsPasswordReset,
+            currentSecret,
+        }: {
+            userId: string;
```

Comment:
it would improve readability if you extracted this type to somewhere above the component.  for example, you could put this before the <UserProfile/> function is defined:

```
type MutationArgs = {
    userId: string;
    currentSecret?: string;
    secret: string;
    needsPasswordReset: boolean;
};
```

then, you can update your declaration of this anonymous func that gets passed into useMutation to be of type MutationArgs.  something like this:

```
({ userId, secret, needsPasswordReset, currentSecret }: MutationArgs) =>
            apiClient.putUserAuthSecret(userId, {
                current_secret: currentSecret,
                needs_password_reset: needsPasswordReset,
                secret: secret,
            }),
```

Reply:
I think i found an even cleaner way, lmk what you think!